docs/pages/docs/routing/middleware.mdx

```diff
@@ -470,7 +470,7 @@ const authMiddleware = withAuth(
 
 export default function middleware(req: NextRequest) {
   const publicPathnameRegex = RegExp(
-    `^(/(${locales.join('|')}))?(${publicPages.join('|')})?/?$`,
+    `^(/(${locales.join('|')}))?(${publicPages.join('|')}|)/?$`,
```

Comment:
Hmm, doesn't this still consider `/` or `/${locale}` public, even if `/` is not included in `publicPages`?

<img width="422" alt="Screenshot 2023-11-17 at 10 17 50" src="https://github.com/amannn/next-intl/assets/4038316/e68c4c6b-dbdc-478e-a34d-77a3c3dbbadc">

Reply:
Looks great!

Note that Next.js by default [removes trailing slashes from pathnames](https://nextjs.org/docs/app/api-reference/next-config-js/trailingSlash). I think we can simplify the code slightly based on that, but other than this, this should be good to go! 👏

Reply:
The only change for trailing slash will be removing the last `/?` which does not simplify a lot. I think we should keep the regex handles both with/without trailing slashes since both are possible.

> By default Next.js will redirect urls with trailing slashes to their counterpart without a trailing slash... **You can configure this behavior to act the opposite way**...

 or do you mean other simplification?